client/my-sites/email/form/mailboxes/components/form.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button } from '@automattic/components';
+import { translate, useRtl } from 'i18n-calypso';
+import { useState } from 'react';
+import { MailboxForm } from 'calypso/my-sites/email/form/mailboxes';
+import { MailboxField } from 'calypso/my-sites/email/form/mailboxes/components/field';
+import {
+	EmailProvider,
+	FormFieldNames,
+	GoogleMailboxFormFields,
+	MailboxFormFieldBase,
+	TitanMailboxFormFields,
+} from 'calypso/my-sites/email/form/mailboxes/types';
+
+interface MailboxFormWrapperProps {
+	domains?: string[];
+	onSubmit: ( mailbox: MailboxForm< EmailProvider > ) => void;
+	provider: EmailProvider;
+	selectedDomainName: string;
+}
+
+const MailboxFormWrapper = ( {
+	children,
+	domains = [],
```

Comment:
I'm wondering why domain should be an optional parameter. I would say that it's mandatory for our case.

```suggestion
	domains,
```

Reply:
Not really. We use `selectedDomainName` more often, and then use `domains` when we need to support sites with multiple domains that can switch from one domain to the other.

That prop then becomes redundant if the site has only one domain and the `selectedDomainName` is always set.

Since it's not being used in this implementation anyways, I'll remove it and add it in another PR.